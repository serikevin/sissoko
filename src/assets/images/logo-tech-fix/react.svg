<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 60.9 54.2"><defs><style>.cls-1{fill:#fff;}</style></defs><path class="cls-1" d="M130.7,26.9c0-4-5.1-7.9-12.8-10.2,1.7-7.9,1-14.2-2.5-16.2a5.3,5.3,0,0,0-2.8-.7V2.6a3.2,3.2,0,0,1,1.4.3c1.7,1,2.4,4.6,1.8,9.4a22.5,22.5,0,0,1-.6,3.6,59.2,59.2,0,0,0-7.9-1.3,54.4,54.4,0,0,0-5.1-6.2c4-3.8,7.8-5.8,10.4-5.8V-.2c-3.4,0-7.9,2.4-12.4,6.6-4.5-4.1-9-6.6-12.4-6.6V2.6c2.6,0,6.4,2.1,10.4,5.8a67.6,67.6,0,0,0-5.1,6.2,59.2,59.2,0,0,0-7.9,1.3,22.5,22.5,0,0,1-.6-3.6c-.6-4.7.1-8.4,1.8-9.4a2.9,2.9,0,0,1,1.4-.3V-.1A5.3,5.3,0,0,0,85,.6c-3.5,2-4.3,8.2-2.5,16.1-7.7,2.4-12.7,6.2-12.7,10.2s5,7.9,12.8,10.2c-1.8,7.9-1,14.2,2.5,16.2a5.1,5.1,0,0,0,2.8.7c3.4,0,7.9-2.4,12.4-6.7,4.5,4.2,9,6.6,12.4,6.6a5.5,5.5,0,0,0,2.8-.7c3.5-2,4.2-8.2,2.4-16.1,7.7-2.4,12.8-6.2,12.8-10.2Zm-16.2-8.3a47.5,47.5,0,0,1-1.7,4.9c-.5-1-1-2-1.6-3l-1.8-2.9,5.1,1Zm-5.7,13.2a38.8,38.8,0,0,1-3,4.8l-5.6.2-5.6-.2-3-4.7c-.9-1.7-1.8-3.3-2.5-5s1.6-3.3,2.5-4.9,2-3.3,3-4.8l5.6-.2,5.6.2q1.5,2.3,3,4.8c.9,1.6,1.8,3.2,2.6,4.9l-2.6,4.9Zm4-1.6a32.8,32.8,0,0,1,1.7,5l-5.1,1,1.8-3c.6-1,1.1-2,1.6-3ZM100.2,43.5c-1.1-1.2-2.3-2.6-3.4-4h6.9C102.5,40.9,101.4,42.3,100.2,43.5ZM91,36.2l-5.1-1c.5-1.6,1.1-3.3,1.7-4.9l1.6,3Zm9.2-25.9c1.1,1.2,2.3,2.6,3.4,4H96.8A44.7,44.7,0,0,1,100.2,10.3ZM91,17.6l-1.8,3-1.6,3c-.7-1.7-1.2-3.4-1.7-5l5.1-1ZM79.8,33.2c-4.4-1.9-7.2-4.4-7.2-6.3s2.8-4.4,7.2-6.3l3.4-1.2A70.1,70.1,0,0,0,86,26.9a52.6,52.6,0,0,0-2.7,7.5,28.6,28.6,0,0,1-3.5-1.2Zm6.7,17.7c-1.7-1-2.5-4.7-1.9-9.4.1-1.2.4-2.4.6-3.6a62,62,0,0,0,7.9,1.3,54.9,54.9,0,0,0,5.2,6.2c-4.1,3.8-7.9,5.8-10.4,5.8a2.6,2.6,0,0,1-1.4-.3Zm29.4-9.5c.6,4.8-.2,8.5-1.8,9.5a3.3,3.3,0,0,1-1.5.3c-2.5,0-6.3-2.1-10.4-5.8a45.6,45.6,0,0,0,5.1-6.2,58.8,58.8,0,0,0,7.9-1.4C115.5,39.1,115.7,40.3,115.9,41.4Zm4.8-8.2-3.5,1.2a53.6,53.6,0,0,0-2.8-7.5,53.6,53.6,0,0,0,2.8-7.5,35.9,35.9,0,0,1,3.5,1.2c4.4,1.9,7.2,4.4,7.2,6.3s-2.9,4.4-7.2,6.3Z" transform="translate(-69.8 0.2)"/><circle class="cls-1" cx="30.4" cy="27.1" r="5.7"/></svg>