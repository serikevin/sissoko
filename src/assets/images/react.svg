<svg id="Layer_1" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 60.9 54.2"><defs><style>.cls-1{fill:#00d8ff;}</style></defs><path class="cls-1" d="M60.9,27.1c0-4-5.1-7.9-12.8-10.2C49.8,9,49.1,2.7,45.6.7A5.5,5.5,0,0,0,42.8,0V2.8a3.2,3.2,0,0,1,1.4.3c1.7,1,2.4,4.6,1.8,9.4a22.5,22.5,0,0,1-.6,3.6,68.5,68.5,0,0,0-7.9-1.3,43.1,43.1,0,0,0-5.1-6.2c4-3.8,7.8-5.8,10.4-5.8V0c-3.4,0-7.9,2.4-12.4,6.6C25.9,2.5,21.4,0,18,0V2.8c2.6,0,6.4,2.1,10.4,5.8a76.4,76.4,0,0,0-5.1,6.2,68.5,68.5,0,0,0-7.9,1.3,22.5,22.5,0,0,1-.6-3.6c-.6-4.7.1-8.4,1.8-9.4A3.2,3.2,0,0,1,18,2.8V.1a5.5,5.5,0,0,0-2.8.7c-3.5,2-4.3,8.2-2.5,16.1C5,19.3,0,23.1,0,27.1S5,35,12.8,37.3c-1.8,7.9-1,14.2,2.5,16.2a5.3,5.3,0,0,0,2.8.7c3.4,0,7.9-2.4,12.4-6.7,4.5,4.2,9,6.6,12.4,6.6a5.5,5.5,0,0,0,2.8-.7c3.5-2,4.2-8.2,2.4-16.1,7.7-2.4,12.8-6.2,12.8-10.2ZM44.7,18.8A40.5,40.5,0,0,1,43,23.7c-.5-1-1-2-1.6-3l-1.8-2.9ZM39,32a38.8,38.8,0,0,1-3,4.8l-5.6.2-5.6-.2-3-4.7c-.9-1.7-1.8-3.3-2.5-5s1.6-3.3,2.5-4.9,2-3.3,3-4.8l5.6-.2,5.6.2q1.5,2.3,3,4.8c.9,1.6,1.8,3.2,2.6,4.9Zm4-1.6a29.5,29.5,0,0,1,1.7,5l-5.1,1,1.8-3C42,32.4,42.5,31.4,43,30.4ZM30.4,43.7c-1.1-1.2-2.3-2.6-3.4-4h6.9C32.7,41.1,31.6,42.5,30.4,43.7Zm-9.2-7.3-5.1-1c.5-1.6,1.1-3.3,1.7-4.9l1.6,3Zm9.2-25.9c1.1,1.2,2.3,2.6,3.4,4H27C28.1,13.1,29.2,11.8,30.4,10.5Zm-9.2,7.3-1.8,3-1.6,3c-.7-1.7-1.2-3.4-1.7-5ZM10,33.4C5.6,31.5,2.8,29,2.8,27.1s2.8-4.4,7.2-6.3l3.4-1.2a60.9,60.9,0,0,0,2.8,7.5,47.8,47.8,0,0,0-2.7,7.5A20.6,20.6,0,0,1,10,33.4Zm6.7,17.7c-1.7-1-2.5-4.7-1.9-9.4.1-1.2.4-2.4.6-3.6a68.5,68.5,0,0,0,7.9,1.3,51.7,51.7,0,0,0,5.2,6.2c-4.1,3.8-7.9,5.8-10.4,5.8A3.2,3.2,0,0,1,16.7,51.1Zm29.4-9.5c.6,4.8-.2,8.5-1.8,9.5a4,4,0,0,1-1.5.3c-2.5,0-6.3-2.1-10.4-5.8a43.1,43.1,0,0,0,5.1-6.2A68.6,68.6,0,0,0,45.4,38C45.7,39.3,45.9,40.5,46.1,41.6Zm4.8-8.2-3.5,1.2a48.1,48.1,0,0,0-2.8-7.5,48.1,48.1,0,0,0,2.8-7.5,20.6,20.6,0,0,1,3.5,1.2c4.4,1.9,7.2,4.4,7.2,6.3S55.2,31.5,50.9,33.4Z" transform="translate(0 0)"/><circle class="cls-1" cx="30.4" cy="27.1" r="5.7"/></svg>