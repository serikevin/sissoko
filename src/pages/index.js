import React from 'react';
import { Layout } from 'components/layout';
import { About } from 'components/about';
import { Ability } from 'components/ability';
import { Portfolio } from 'components/portofolio';
import Sertification from 'components/certificate';
import { HireMe } from 'components/hireme';

export const Home = () => {
    return (
        <Layout>
            <About />
            <Ability />
            <Portfolio />
            <Sertification />
            <HireMe />
        </Layout>
    );
};