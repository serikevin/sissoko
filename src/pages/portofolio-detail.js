import React, { Fragment, useEffect, useState } from 'react'
import { useParams } from "react-router-dom";
import { Carousel } from 'react-responsive-carousel';
import "react-responsive-carousel/lib/styles/carousel.min.css";
import ContentLoader from "react-content-loader"
import axios from 'axios';
import { SERVER_IMAGE_URL, SERVER_URL } from 'constant/url';

export const PortfolioDetail = () => {
    let { id } = useParams();

    const [DataResponse, setDataResponse] = useState(null);

    useEffect(() => {
        axios.get(SERVER_URL + '/portofolio/detail/' + id).then(function (response) {
            if (response.data.status === true) {
                setDataResponse(response.data.body);
            }
        }).catch(function (error) {
            
        }).then(function () {
            
        }); 
        return () => {
            setDataResponse(null);
        };
    }, [id]);

    return (
        <Fragment>
            {(DataResponse != null) ?
                <section className="bg-dark pt140 pb50 text-left">
                    <div className="container">
                        <div className="row">
                            <div className="col-md-12 text-center">
                                <h2 className='text-white'><strong>{DataResponse.title}</strong></h2>
                                <p className="lead">{DataResponse.descriptions}</p>
                            </div>
                            <div className="col-md-8 mt20 mb40">
                                <div className="carousel lightbox" data-autoplay="false" data-speed={4000} data-touch-drag="true" data-loop="false">
                                <Carousel showStatus={false} showIndicators={false}>
                                    {
                                        DataResponse.images && DataResponse.images.map((item, index) => {
                                            return (
                                                <div key={index}>
                                                    <img  alt={DataResponse.title} src={SERVER_IMAGE_URL + item.images} />
                                                </div>
                                            )
                                        })
                                    }
                                    </Carousel>
                                </div>
                            </div>
                            <div className="col-md-4 project-sidebar">
                                <div>
                                    <h4 className='text-white'><strong>Project Info</strong></h4>
                                    <p>{DataResponse.project_info}</p>
                                </div>
                                <div className="project-info">
                                    <div>
                                        <p className='text-white'>Platform</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">{DataResponse.platform}</span>
                                    </div>
                                    <div>
                                        <p className='text-white'>Programming Language</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">{DataResponse.languages}</span>
                                    </div>
                                    <div>
                                        <p className='text-white'>Database</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">{DataResponse.databases}</span>
                                    </div>
                                    <div>
                                        <p className='text-white'>Tanggal</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">{DataResponse.dates}</span>
                                    </div>
                                    <div>
                                        <p className='text-white'>Url</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">
                                            <a target={'_blank'} rel="noreferrer" href={DataResponse.urls}>
                                                {DataResponse.urls}
                                            </a>
                                        </span>
                                    </div>
                                    <div>
                                        <p className='text-white'>Source Code</p>
                                    </div>
                                    <div className="mbb-2">
                                        <span className="desc-portfolio">
                                            {
                                                (DataResponse.source_code.includes("http") || DataResponse.source_code.includes("https")) ?
                                                    <a target={'_blank'} rel="noreferrer" href={DataResponse.source_code}>
                                                        {DataResponse.source_code}
                                                    </a>
                                                : DataResponse.source_code
                                            }
                                        </span>
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </section>
                : 
                <section className="bg-dark pb50 text-left">
                     <div className="container">
                        <div className="row">
                <ContentLoader
                    viewBox="0 0 700 300"
                    backgroundColor="#111111"
                    foregroundColor="#000000"
                >
                   
                    <rect x="376" y="253" rx="14" ry="14" width="72" height="32" />

                    <rect x="55" y="42" rx="16" ry="16" width="274" height="416" />
                    <rect x="376" y="101" rx="3" ry="3" width="231" height="10" />
                    <rect x="376" y="121" rx="3" ry="3" width="150" height="10" />
                    <rect x="376" y="81" rx="3" ry="3" width="231" height="10" />
                    <rect x="376" y="41" rx="3" ry="3" width="231" height="29" />

                    <rect x="376" y="140" rx="3" ry="3" width="231" height="29" />
                    <rect x="376" y="180" rx="3" ry="3" width="150" height="10" />
                    <rect x="376" y="200" rx="3" ry="3" width="100" height="10" />
                    <rect x="376" y="220" rx="3" ry="3" width="150" height="10" />
                </ContentLoader>
                </div>
                </div>
                </section>
            }
        </Fragment>
    )
};