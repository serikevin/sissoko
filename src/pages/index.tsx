import About from 'components/About';
import MainButton from 'components/MainButton';
import SocialButton from 'components/SocialButton';
import AbilityCard from 'components/AbilityCard';
import UniqueCard from 'components/UniqueCard';
import PortfolioCard from 'components/PortfolioCard';
import CertificateCard from 'components/CertificateCard';
import { DEFAULT_PROFILE_IMAGE } from 'constant/images';
import { DEFAULT_CV_DOCUMENT } from 'constant/document';
import { DEFAULT_AWESOME_AMAZING_JSON, DEFAULT_CLEAN_DESIGN_JSON, DEFAULT_COMPLEXITY_JSON } from 'constant/json';

export const Home = () => {
    return (
        <>
            <section className="hero-fullscreen bg-main parallax">
                <div className="container css-nf_gi4ga">
                    <div className="css-jge_cvjw">
                        <div className="css-fgE_c">
                            <h3 className='text-highlight hello-for-responsive'>Hello, I am</h3>
                            <h1 className='font-bold text-title pb-0 main-title-res'>Yovie Alfa Guistuta</h1>
                            <About isFooter={false} />
                            <div className="button-main-about">
                                <MainButton links={DEFAULT_CV_DOCUMENT} text={"Download Resume"} isFirst={true} />
                                <MainButton links={"mailto:yoviealfaguistuta@gmail.com"} text={"Contact Me"} isFirst={false} />
                            </div>
                        </div>
                        <div className="text-center">
                            <img className='image-about' src={DEFAULT_PROFILE_IMAGE} alt="Yovie Alfa Guistuta" />
                        </div>
                    </div>
                </div>
                <div className="social-bottom-icon">
                    <SocialButton />
                </div>
            </section>
            <section className="pt100 bg-dark mt-cs pb100">
                <div className="background-image">
                    <img src="img/backgrounds/bg-2.jpg" alt="#" />
                </div>
                <div className="container">
                    <div className="col-md-12 text-center pb20">
                        <h2 className='text-title'>Skills and Tech Stack</h2>
                        <p className="text-desc">My <span className="text-highlight">main skills and tech stack</span> when developing system based on here .</p>
                    </div>
                    <div className="row vertical-align">
                        <div className="col-md-12 pt40 pb30">
                            <div className="progress-bars standard transparent-bars">
                                <div className="row">
                                    <div className="col-md-6">
                                        <AbilityCard title={'Frontend Developer'} description={'Develop apps from mockups into code or use my creativity to create awesome UI and still make the UX super easy to use'} icon={['react', 'laravel', 'jquery']} />
                                    </div>
                                    <div className="col-md-6">
                                        <AbilityCard title={'Backend Developer'} description={'Develop API with highly efficiently and make Frontend easy to consume with nice response, complete API documentation and always care about security'} icon={['laravel', 'go']} />
                                    </div>
                                </div>
                            </div>
                        </div>
                    </div>
                </div>
            </section>
            <section className="parallax mt-res-unique bg-dark pb90">
                <div className="container">
                    <div className="row">
                        <div className="col-md-12 text-center pb20">
                            <h2 className='text-title'>Unique and Strength</h2>
                            <p className="text-desc">My <span className="text-highlight">unique and strength</span> when developing a system</p>
                        </div>
                        <div className="col-md-12">
                            <div className="testimonials quote-icons centered-cd" data-autoplay="false" data-speed={4000}>
                                <UniqueCard json={DEFAULT_AWESOME_AMAZING_JSON} title={'Awesome and Amazing'} description={'When I make a program then I will make it <span className="text-highlight">as good as possible</span> because this is my work<br /> and people will use what I have created. A good rating from them is what I look forward to the most.<i className="vossen-quote text-white" />'} />
                                <UniqueCard json={DEFAULT_COMPLEXITY_JSON} title={'Complexity'} description={'Making programs that are quite detailed is what I like the most. <br />A program with <span className="text-highlight">very complete features</span> makes the results that are done more satisfying<i className="vossen-quote text-white" />'} />
                                <UniqueCard json={DEFAULT_CLEAN_DESIGN_JSON} title={'Clean Design'} description={"The user interface is something that needs to be <span className='text-highlight'>considered</span> in making a program. <br />I don't like the appearance of a crowded or messy program because it's very uncomfortable to look at.<i className='vossen-quote text-white' />"} />
                            </div>
                        </div>
                    </div>
                </div>
            </section>

            <section className="bg-main pt100 pb110">
                <div className="text-center">
                    <div className="text-center pb20">
                        <h2 className='text-title'>My Greatest Portfolio</h2>
                        <p className="text-desc">The following is the system that I have made so far. It will <span className="text-highlight">keep growing</span></p>
                    </div>
                    <div className="container">
                        <PortfolioCard />
                    </div>
                </div>
            </section>

            <section className="bg-dark pt100 pb110">
                <div className="container">
                    <div className="text-center">
                        <div className="text-center pb20">
                            <h2 className='text-title'>My Certificate and Awards</h2>
                            <p className="text-desc">There is <span className="text-highlight">competitions</span>, <span className="text-highlight">training</span> and <span className="text-highlight">workshops</span>.</p>
                        </div>
                        <CertificateCard />
                    </div>
                </div>
            </section>
            <section className="container bg-main pt100 pb110">
                <div className="container">
                    <div className="text-center">
                        <h2 className="text-title">Ready to <span className="bold text-highlight">Kickstart</span> Your Plans ?..</h2>
                        <p className="lead res-lead-kick"><span className="text-highlight">Hire me</span> and i will give you the <span className="text-highlight">best experience</span></p>
                        <div className="button-main-about">
                            <MainButton links={DEFAULT_CV_DOCUMENT} text={"Download Resume"} isFirst={true} />
                            <MainButton links={"mailto:yoviealfaguistuta@gmail.com"} text={"Contact Me"} isFirst={false} />
                        </div>
                    </div>
                </div>
            </section>
        </>
    );
};