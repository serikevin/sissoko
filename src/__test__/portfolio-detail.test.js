import { render } from '@testing-library/react';
import { PortfolioDetail } from 'pages/portofolio-detail';

test('Render Portfolio Detail', () => {
	const { container } = render(<PortfolioDetail />);
	expect(container).toMatchSnapshot()
});
