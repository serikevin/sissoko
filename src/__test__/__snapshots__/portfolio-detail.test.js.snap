// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Portfolio Detail 1`] = `
<div>
  <section
    class="bg-dark pb50 text-left"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <svg
          aria-labelledby="vj0ycm-aria"
          role="img"
          viewBox="0 0 700 300"
        >
          <title
            id="vj0ycm-aria"
          >
            Loading...
          </title>
          <rect
            clip-path="url(#vj0ycm-diff)"
            height="100%"
            role="presentation"
            style="fill: url(#vj0ycm-animated-diff);"
            width="100%"
            x="0"
            y="0"
          />
          <defs>
            <clippath
              id="vj0ycm-diff"
            >
              <rect
                height="32"
                rx="14"
                ry="14"
                width="72"
                x="376"
                y="253"
              />
              <rect
                height="416"
                rx="16"
                ry="16"
                width="274"
                x="55"
                y="42"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="101"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="121"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="81"
              />
              <rect
                height="29"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="41"
              />
              <rect
                height="29"
                rx="3"
                ry="3"
                width="231"
                x="376"
                y="140"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="180"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="100"
                x="376"
                y="200"
              />
              <rect
                height="10"
                rx="3"
                ry="3"
                width="150"
                x="376"
                y="220"
              />
            </clippath>
            <lineargradient
              id="vj0ycm-animated-diff"
            >
              <stop
                offset="0%"
                stop-color="#111111"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-2; -2; 1"
                />
              </stop>
              <stop
                offset="50%"
                stop-color="#000000"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="-1; -1; 2"
                />
              </stop>
              <stop
                offset="100%"
                stop-color="#111111"
                stop-opacity="1"
              >
                <animate
                  attributeName="offset"
                  dur="1.2s"
                  keyTimes="0; 0.25; 1"
                  repeatCount="indefinite"
                  values="0; 0; 3"
                />
              </stop>
            </lineargradient>
          </defs>
        </svg>
      </div>
    </div>
  </section>
</div>
`;
